### PR TITLE
networkd: ndisc - add support for prefix allow-list and route allow-list

### PR DESCRIPTION
**Pull request by ssahani**
It requires a testing /. will do that.

### CODE REVIEW

File: src/network/networkd-ndisc.c [hunk 1]
```diff
@@ -1445,6 +1459,75 @@ int config_parse_ndisc_deny_listed_prefix(
...
+
+                r = set_ensure_consume(list, &in6_addr_hash_ops, TAKE_PTR(a));
+                if (r < 0)
+                        return log_oom();
+                if (r == 0)
+                        log_syntax(unit, LOG_WARNING, filename, line, 0,
+                                   "NDisc allow-listed %sprefix entry %s is duplicated, ignoring assignment.",
+                                   is_route ? "route " : "", n);
+                if (r > 0)
+                        TAKE_PTR(a);
```
Comment (bluca) [inline src/network/networkd-ndisc.c]: TAKE_PTR(a) is repeated?

**Comment by bluca** [comment]
> It requires a testing /. will do that.

Have you done the testing now?

File: src/network/networkd-ndisc.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-ndisc.c]: Ouch! This is my fault in the previous commit...

**Comment by ssahani** [comment]
> > It requires a testing /. will do that.
> 
> Have you done the testing now?

On vacation goa india away from kb. Sorry will do .

**Comment by bluca** [comment]
> > > It requires a testing /. will do that.
> > 
> > 
> > Have you done the testing now?
> 
> On vacation goa india away from kb. Sorry will do .

No problem at all and no rush, just checking. Enjoy your vacation!

File: src/network/networkd-ndisc.c (hunk 1 above)
Comment (yuwata) [reply src/network/networkd-ndisc.c]: Fixed.

**Comment by yuwata** [comment]
I force-pushed to add a fix for memleak. I will also add a test case for this.

**Comment by yuwata** [comment]
@ssahani and @bluca I added several tests for the new settings. While that, I found several issues, and the fixes for them are also included in this PR now. Furthermore, I added `RouterAllowList=` and `RouterDenyList=`. PTAL.

**Review by bluca — APPROVED** [review]

**Comment by ssahani** [comment]
Thanks @bluca  and @yuwata

**Pull request merged**